aries_cloudagent/core/conductor.py

```diff
@@ -311,26 +312,41 @@ async def start(self) -> None:
         )
 
         # record ACA-Py version in Wallet, if needed
+        from_version = None
+        agent_version = f"v{__version__}"
         async with self.root_profile.session() as session:
             storage: BaseStorage = session.context.inject(BaseStorage)
-            agent_version = f"v{__version__}"
             try:
                 record = await storage.find_record(
                     type_filter=RECORD_TYPE_ACAPY_VERSION,
                     tag_query={},
                 )
-                if record.value != agent_version:
-                    LOGGER.exception(
-                        (
-                            f"Wallet storage version {record.value} "
-                            "does not match this ACA-Py agent "
-                            f"version {agent_version}. Run aca-py "
-                            "upgrade command to fix this."
-                        )
-                    )
-                    raise
+                from_version = record.value
             except StorageNotFoundError:
-                pass
+                LOGGER.exception(("Wallet version storage record not found."))
+        from_version = from_version or self.root_profile.settings.get(
+            "upgrade.config_path"
+        )
+        if from_version:
+            config_available_list = get_upgrade_version_list(
+                config_path=self.root_profile.settings.get("upgrade.config_path"),
+                from_version=from_version,
+            )
+            if len(config_available_list) >= 1 and (
+                from_version != agent_version
+                or self.root_profile.settings.get("upgrade.force_upgrade")
+            ):
+                await upgrade(self.root_profile.settings)
+        else:
+            LOGGER.exception(
```

Comment:
This looks to me like if the wallet version is not found in storage, an exception is raised and the app stops — is that right?  If so, that is not what we want.  If the user does not pass a `—from-version`, and there is no `from` in storage, then I think the app should write the current version to storage, so that it will be found in the future.  A note should be written to the log with instructions on what to do if this is unexpected (notably — how to run the upgrade from the command line).

If the user then determines that they `upgrade` was bypassed, then they can run `—from-version x.y.z` and `—force-upgrade` to apply the upgrade.